core/src/main/java/org/elasticsearch/ElasticsearchRemoteException.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * Exception thrown when a remote action fails. Allows settings an arbitrary {@link RestStatus} so the remote failures that include status
+ * information can be returned without losing anything.
+ */
+public class ElasticsearchRemoteException extends ElasticsearchException {
```

Comment:
this exception is oddly similar (if not the same) to ElasticsearchSecurityException. I guess you didn't want to use that one due to the naming. I guess if we should rename that other one. ElasticsearchRemoteException is a good and specific name for this specific usecase but it's only going to be used here.